tsdb/engine/tsm1/engine.go

```diff
@@ -1497,6 +1492,27 @@ func (e *Engine) deleteSeriesRange(seriesKeys [][]byte, min, max int64) error {
 		max = math.MaxInt64
 	}
 
+	overlapsTimeRangeMinMax := false
+	e.FileStore.Apply(func(r TSMFile) error {
+		if r.OverlapsTimeRange(min, max) {
+			overlapsTimeRangeMinMax = true
+		}
+		return nil
+	})
+
+	if !overlapsTimeRangeMinMax && e.Cache.store.count() > 0 {
```

Comment:
Ideally we could check the timerange on `e.Cache`, but checking for non-emptiness is the best I found.